### ISSUE TITLE
peerRouting.findPeer() does not accept strings, only PeerId instances

### ISSUE THREAD

**Issue by a1300** [libp2p/js-libp2p#819]
- **Version**: v0.29.3
- **Platform**: Linux
- **Subsystem**:

#### Description: `peerRouting.findPeer()` does not accept string representations for `PeerId`, it only accepts `PeerId` instances

The [`API.md#peerroutingfindpeer`](https://github.com/libp2p/js-libp2p/blob/master/doc/API.md#peerroutingfindpeer) says `findPeer()` only accepts `PeerId` instances

On the other hand the `JSDoc` type documentation `findPeer()` only accepts `string`

https://github.com/libp2p/js-libp2p/blob/e9e4b731a5be2a3c25506d4edf4a2ef84b748e3c/src/peer-routing.js#L14-L23

If I understand the following test correctly, it uses `findPeer()` also with a `string`

https://github.com/libp2p/js-libp2p/blob/e9e4b731a5be2a3c25506d4edf4a2ef84b748e3c/test/peer-routing/peer-routing.node.js#L115-L129

#### Steps to reproduce the error:

```bash
git clone https://gist.github.com/a1300/79357f742a6e459a9a801e60bd2840ef libp2p-peerRouting-no-string
cd libp2p-peerRouting-no-string
npm install
node index
```

**Comment by vasco-santos** [comment]
Per https://github.com/libp2p/js-libp2p-interfaces/tree/master/src/peer-routing#api this should definitely be `PeerId` and not string.  Both the DHT and delegated peer routing also expect `PeerId`.

The test mocks the response as the "real" tests for the delegate are in the delegate repo. As a result, what is provided in the parameter is not even used. Anyway, we should turn it into a PeerId with `PeerId.createFromCID`. Want to PR these changes?

**Comment by vasco-santos** [comment]
Also, FYI I am working on #802 . Once we get all the modules to use this, these type of documentation errors should be caught in CI

**Comment by vasco-santos** [comment]
Just shipped libp2p rc for `0.30` where this is fixed. Type checking will now run.
